src/osd/ECBackend.cc

```diff
@@ -2142,38 +2142,42 @@ void ECBackend::check_ops()
 }
 
 int ECBackend::objects_read_sync(
-  const hobject_t &hoid,
+  const hobject_t &soid,
+  const shard_id_t &shard,
   uint64_t off,
   uint64_t len,
   uint32_t op_flags,
-  bufferlist *bl)
+  unsigned size,
+  bufferlist *bl,
+  boost::optional<uint32_t> maybe_crc,
+  bool sparse)
 {
   return -EOPNOTSUPP;
 }
 
 void ECBackend::objects_read_async(
-  const hobject_t &hoid,
-  const list<pair<boost::tuple<uint64_t, uint64_t, uint32_t>,
```

Comment:
:+1: for dissecting this into `ReadItem`. The pair of tuple and another pair was really hard to read. It would be really beneficial to put clean-ups like this into separated commits.